build_all/make_sdk.py

```diff
@@ -0,0 +1,192 @@
+import os
+import shutil
+from distutils import dir_util
+import sys
+import getopt
+import glob
+import make_sdk_cmds_dict as commands_dict
+
+def pattern_copy(pattern): # helper method
+    # Get a list of all the files that match the pattern in specified directory
+    fileList = glob.glob(pattern)
+    for filePath in fileList:
+        try:
+            shutil.copy2(filePath)
+        except:
+            print("Error while copying file : ", filePath)
+
+
+def pattern_delete(pattern): # helper method
+    # Get a list of all the files that match the pattern in specified directory
+    fileList = glob.glob(pattern)
+    # Iterate over the list of filepaths & remove each file.
+    for filePath in fileList:
+        try:
+            os.remove(filePath)
+        except:
+            print("Error while deleting file : ", filePath)
+
+
+def pattern_delete_folder(pattern): # helper method
+    # Get a list of all the file paths that match the pattern in specified directory
+    fileList = glob.glob(pattern)
+    # Iterate over the list of filepaths & remove each file.
+    for filePath in fileList:
+        try:
+            shutil.rmtree(filePath)
+        except:
+            print("Error while deleting filepath : ", filePath)
+
+def line_prepender(filename, line):
+    with open(filename, 'r+') as f:
+        content = f.read()
+        f.seek(0, 0)
+        f.write(line.rstrip('\r\n') + '\n' + content)
+
+def line_appender(filename, line):
+    with open(filename, 'a') as f:
+        f.write(line.rstrip('\r\n') + '\n')
+
+def usage():
+    # Iterates through command dictionary to print out script's opt usage
+    usage_txt = "make_sdk.py accepts the following arguments: \r\n"
+
+    for commands in commands_dict.cmds:
+        usage_txt += " - %s: " %commands + commands_dict.cmds[commands]['text'] + "\r\n"
+
+    return usage_txt
+
+def parse_opts():
+    options, remainder = getopt.gnu_getopt(sys.argv[1:], 'ho:d:', ['output', 'help', 'device'])
+    # print('OPTIONS   :', options)
+    for opt, arg in options:
+        if opt in ('-h', '--help'):
+            print(usage())
+        elif opt in ('-o', '--output'):
+            commands_dict.output_path = arg
+            
+
+def run():
+    # set up paths for copying
+    output_path = os.path.abspath(commands_dict.output_path)
+    arduino_repo_root = os.path.abspath('../')
+    print(output_path)
+    arduino_pal_path = arduino_repo_root+'/pal/'
+    azure_iot_sdk_path = arduino_repo_root+'/sdk/'
+    AzureIoTHub_path = output_path+'/AzureIoTHub/'
+    AzureIoTProtocolHTTP_path = output_path+'/AzureIoTProtocol_HTTP/'
```

Comment:
>+ [](start = 43, length = 1)

nit: can you add space before and after +.

Reply:
Done.